x-pack/legacy/plugins/siem/index.ts

```diff
@@ -125,17 +123,36 @@ export const siem = (kibana: any) => {
       mappings: savedObjectMappings,
     },
     init(server: Server) {
-      const newPlatform = ((server as unknown) as KbnServer).newPlatform;
-      if (server.plugins.alerting != null) {
-        const type = signalsAlertType({
-          logger: newPlatform.coreContext.logger.get('plugins', APP_ID),
-        });
-        if (isAlertExecutor(type)) {
-          server.plugins.alerting.setup.registerType(type);
-        }
-      }
-      server.injectUiAppVars('siem', async () => server.getInjectedUiAppVars('kibana'));
-      initServerWithKibana(server);
+      const {
+        config,
+        getInjectedUiAppVars,
+        indexPatternsServiceFactory,
+        injectUiAppVars,
+        newPlatform,
+        plugins,
+        route,
+        savedObjects,
+      } = server;
+
+      const { logger } = newPlatform.coreContext;
+
+      const serverFacade = {
+        config,
+        getInjectedUiAppVars,
+        indexPatternsServiceFactory,
+        injectUiAppVars,
```

Comment:
you're using `injectUiAppVars ` and `getInjectedUiAppVars` to inject the kibana plugins app vars into the browser: https://github.com/elastic/kibana/blob/d8901374cffedd569a2ff2f0af8e5e298f008caf/src/legacy/core_plugins/kibana/inject_vars.js#L31-L35

The NP won't allow you to inject vars into the browser anymore, so these values will either have to come from other plugins that might expose them on the browser, or by making a get request to the server. Are you using any of these values on the client side?

(We will expose a browser-side config service https://github.com/elastic/kibana/issues/41990 but this is only for configuration values "owned" by your plugin so this doesn't seem relevant to SIEM's server-side plugin.)

Reply:
We weren't using these, so that was an easy migration 😉